src/hashmap.c

```diff
@@ -1,51 +1,36 @@
-#define BUCKETS 1000000
-#define PRIME_SEED_1 17
-#define PRIME_SEED_2 97
-#define KEY_CHARS 128
-#define VALUE_CHARS 128
-#define NODE_SIZE sizeof(Node)
-
+#include "hashmap.h"
 #include <stdlib.h>
-#include <stdio.h>
 #include <stdbool.h>
 #include <string.h>
-#include <assert.h>
-
-/* A key-value node in a single-linked list. */
-typedef struct Node Node;
-
-struct Node {
-    Node* next;
-    char key[KEY_CHARS];
-    char value[VALUE_CHARS]; 
-};
 
+#define BUCKETS 1000000
+#define PRIME_SEED_1 17
+#define PRIME_SEED_2 97
 
+/* A key-value node in a single-linked list. */
 Node* create_sentinel_node(){
-    void* ptr = malloc(NODE_SIZE);
```

Comment:
void* can be implicitly casted to any other pointer type, so there is no need for the `Node* node = (Node*)ptr;`

Also you can check if a Node* is NULL just the same way as with a void*. A pointer is just a pointer no matter what type it is pointing to